react/components/BackToTopButton/index.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState, useEffect } from 'react'
+import { Button } from 'vtex.styleguide'
+import { FormattedMessage } from 'react-intl'
+import { Icon } from 'vtex.store-icons'
+import classNames from 'classnames'
+
+import styles from './styles.css'
+
+interface Props {
+  displayThreshold?: number
+  display?: Display
+}
+
+type Display = 'button' | 'caret-icon'
+
+const handleBackTop = () => {
+  window.scroll({ top: 0, left: 0, behavior: 'smooth' })
+}
+
+const BackToTopButton: StorefrontFC<Props> = props => {
+  const [isShowed, setIsShowed] = useState(false)
+  const [scrollY, setScrollY] = useState(0)
+  const backToTopButtonClasses = classNames(
+    `${styles.backToTopButtonContainer} z-999 fixed bottom-2 right-2`,
+    {
+      [`${styles.backToTopButtonHidden}`]: !isShowed,
+      [`${styles.backToTopButtonActive}`]: isShowed,
+    }
+  )
+
+  function scrollValue() {
+    setScrollY(window.pageYOffset)
+  }
+
+  useEffect(() => {
+    function watchScroll() {
+      window.addEventListener('scroll', scrollValue)
+    }
+    watchScroll()
+    return () => {
+      window.removeEventListener('scroll', scrollValue)
+    }
+  })
+
+  useEffect(() => {
+    if (scrollY > props.displayThreshold!) {
```

Comment:
This non-null assertion could be dangerous.
Also, TypeScript does not pick-up the `defaultProps` property you're using below. Just to be safe, move the values from `BackToTopButton.defaultProps` in the component itself, something like:

```tsx
const BackToTopButton: StorefrontFC<Props> = ({ displayThreshold = 600, display = 'button' }) => {
```